nshlib/nsh_printf.c

```diff
@@ -84,7 +87,34 @@ int cmd_printf(FAR struct nsh_vtbl_s *vtbl, int argc, char **argv)
             {
               case 'x':
                 fmt++;
-                nsh_output(vtbl, "%c", strtol(fmt, NULL, 16));
+                value = strtol(fmt, NULL, 16);
+                len = strnlen(fmt, 10);
+
+                /* Is it a Big Endian MCU ? */
+
+                if (BYTE_ORDER == BIG_ENDIAN)
+                  {
+                    value = __swap_uint32(value);
```

Comment:
Don't manually swap the value.
(0x12345678 & 0xff00) >> 8 always get 0x56.

Reply:
Ok! I will remove this check. I need to find a Big Endian MCU supported by NuttX. Doing "git grep "CONFIG_ENDIAN_BIG=y" " inside boards/ reveals some. I will try to get a powered by at32uc3.